File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque

N, M, V = map(int, input().split())
visited = [False for _ in range(N+1)]
vertex = [[] for _ in range(N+1)]

for i in range(1,M+1):
    x,y = map(int, input().split())
    vertex[x].append(y)
    vertex[y].append(x)

for v in vertex:
    v.sort()

def dfs(start):
    visited[start] = True
    print(start, end = " ")

    for v in vertex[start]:
        if not visited[v]:
            dfs(v)

def bfs(start):
    d = deque()
    print(start, end= " ")
    visited[start] = True
    d.append(start)

    while d:
        n = d.popleft()

        for i in vertex[n]:
            if not visited[i]:
                print(i, end=" ")
                visited[i] = True
                d.append(i)

dfs(V)
print()
visited = [False for _ in range(N+1)]
bfs(V)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    static int n;
    static LinkedList<Integer>[] graph;
    static boolean[] visited;

    public static void dfs(int start) {
        System.out.print(start + " ");
        visited[start] = true;

        for (int vertex : graph[start]) {
            if (!visited[vertex]) {
                dfs(vertex);
            }
        }
    }

    public static void bfs(int start) {
        Queue<Integer> queue = new LinkedList<>();
        queue.add(start);
        visited[start] = true;
        System.out.print(start + " ");

        while (!queue.isEmpty()) {
            int temp = queue.poll();
            for (int vertex : graph[temp]) {
                if (!visited[vertex]) {
                    System.out.print(vertex + " ");
                    visited[vertex] = true;
                    queue.add(vertex);
                }
            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer stringTokenizer = new StringTokenizer(bufferedReader.readLine());
        n = Integer.parseInt(stringTokenizer.nextToken());
        graph = new LinkedList[n+1];
        visited = new boolean[n+1];
        int m = Integer.parseInt(stringTokenizer.nextToken());
        int v = Integer.parseInt(stringTokenizer.nextToken());

        for (int i = 1; i <= n; i++) {
            graph[i] = new LinkedList<>();
        }

        for (int i = 0; i < m; i++) {
            stringTokenizer = new StringTokenizer(bufferedReader.readLine());
            int start = Integer.parseInt(stringTokenizer.nextToken());
            int end = Integer.parseInt(stringTokenizer.nextToken());

            graph[start].add(end);
            graph[end].add(start);
        }

        for (int i = 1; i <= n; i++) {
            Collections.sort(graph[i]);
        }

        dfs(v);
        System.out.println("");
        visited = new boolean[n+1];
        bfs(v);
    }
}
